src/server/tran_server.cpp

```diff
@@ -330,49 +329,56 @@ tran_server::connection_handler::connect ()
+    m_state = state::CONNECTING;
 
-  // connect to page server
-  constexpr int CHANNEL_POLL_TIMEOUT = 1000;    // 1000 milliseconds = 1 second
-  cubcomm::server_channel srv_chn (m_ts.m_server_name.c_str (), SERVER_TYPE_PAGE, CHANNEL_POLL_TIMEOUT);
+    // connect to page server
+    constexpr int CHANNEL_POLL_TIMEOUT = 1000;    // 1000 milliseconds = 1 second
+    cubcomm::server_channel srv_chn (m_ts.m_server_name.c_str (), SERVER_TYPE_PAGE, CHANNEL_POLL_TIMEOUT);
 
-  srv_chn.set_channel_name ("TS_PS_comm");
+    srv_chn.set_channel_name ("TS_PS_comm");
 
-  css_error_code comm_error_code = srv_chn.connect (m_node.get_host ().c_str (), m_node.get_port (),
-				   CMD_SERVER_SERVER_CONNECT);
-  if (comm_error_code != css_error_code::NO_ERRORS)
-    {
-      return ps_conn_error_lambda (lockg_state);
-    }
+    css_error_code comm_error_code = srv_chn.connect (m_node.get_host ().c_str (), m_node.get_port (),
+				     CMD_SERVER_SERVER_CONNECT);
+    if (comm_error_code != css_error_code::NO_ERRORS)
+      {
+	return ps_conn_error_lambda (lockg_state);
+      }
 
-  if (srv_chn.send_int (static_cast<int> (m_ts.m_conn_type)) != NO_ERRORS)
-    {
-      return ps_conn_error_lambda (lockg_state);
-    }
+    if (srv_chn.send_int (static_cast<int> (m_ts.m_conn_type)) != NO_ERRORS)
+      {
+	return ps_conn_error_lambda (lockg_state);
+      }
 
-  int returned_code;
-  if (srv_chn.recv_int (returned_code) != css_error_code::NO_ERRORS)
-    {
-      return ps_conn_error_lambda (lockg_state);
-    }
-  if (returned_code != static_cast<int> (m_ts.m_conn_type))
-    {
-      return ps_conn_error_lambda (lockg_state);
-    }
+    int returned_code;
+    if (srv_chn.recv_int (returned_code) != css_error_code::NO_ERRORS)
+      {
+	return ps_conn_error_lambda (lockg_state);
+      }
+    if (returned_code != static_cast<int> (m_ts.m_conn_type))
+      {
+	return ps_conn_error_lambda (lockg_state);
+      }
+
+    // NOTE: only the base class part (cubcomm::channel) of a cubcomm::server_channel instance is
+    // moved as argument below
+    set_connection (std::move (srv_chn));
 
-  // NOTE: only the base class part (cubcomm::channel) of a cubcomm::server_channel instance is
-  // moved as argument below
-  set_connection (std::move (srv_chn));
+    // TODO initailizing jobs will be triggered such as the catch-up process. state::CONNECTED will be set asynchronously when it's done.
 
-  // TODO initailizing jobs will be triggered such as the catch-up process. state::CONNECTED will be set asynchronously when it's done.
+    m_state = state::CONNECTED;
 
-  m_state = state::CONNECTED;
+    er_log_debug (ARG_FILE_LINE, "Transaction server successfully connected to the page server. Channel id: %s.\n",
+		  srv_chn.get_channel_id ().c_str ());
+  }
 
-  er_log_debug (ARG_FILE_LINE, "Transaction server successfully connected to the page server. Channel id: %s.\n",
-		srv_chn.get_channel_id ().c_str ());
+  if (m_ts.reset_main_connection () != NO_ERROR)
```

Comment:
How about reverting changes and add reset_main_connection() at the following position?
```
tran_server::ps_connector::try_connect_to_all_ps (cubthread::entry &)
{
  for (const auto &conn : m_ts.m_page_server_conn_vec)
    {
      if (conn->is_idle ())
        {
          /*
           * TODO It can be too verbose now since it always complain to fail to connect when a PS has been stopped.
           * Later on, this job is going to be triggered by a cluster manager or cub_master when a PS is ready to connect.
           */
          conn->connect ();
--> this position (after new conection is established)
        }

      if (m_terminate.load ())
        {
          break;
        }
    }
}
```

Reply:
It also should be done during boot() because the catch-up information containing main connection's info has to be sent during catch-up in connect().

Reply:
Is it enough to wait if there is no main connection for a while on boot when ATS receives a catch-up request from PS?

Reply:
Please clarify it. ATS doesn't "receive" the request, it sends. and it doesn't need to wait for it. Shortly, it sends dummy information to indicate "you're the target PS" when there is no PS, and sends the main connection information to others. This will be addressed in http://jira.cubrid.org/browse/LETS-746 actually.

Reply:
Discuss off-line. Please add comment about to-do.